common/core/web/keyboard-processor/test.sh

```diff
@@ -0,0 +1,65 @@
+#! /bin/bash
+
+# We should work within the script's directory, not the one we were called in.
+cd $(dirname "$BASH_SOURCE")
+
+# Include useful testing resource functions
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../../resources/build/build-utils.sh"
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# A simple utility script to facilitate unit-testing for the LM Layer.
+# It's rigged to be callable by NPM to facilitate testing during development when in other folders.
+
+display_usage ( ) {
+  echo "test.sh [-skip-package-install] [-CI] [ -? | -h | -help]"
+  echo "  -CI               to perform continuous-integration friendly tests and reporting"
+  echo "  -? | -h | -help   to display this help information"
+  echo "  -skip-package-install         to bypass refreshing dependencies.  Useful when called by scripts that pre-fetch"
+  echo ""
+  exit 0
+}
+
+test-headless ( ) {
+  _FLAGS=$FLAGS
+  if (( CI_REPORTING )); then
+    _FLAGS="$_FLAGS --reporter mocha-teamcity-reporter"
+  fi
+
+  npm run mocha -- --recursive $_FLAGS ./tests/cases/*.js
+}
+
+# Defaults
+get_builder_OS  # return:  os_id="linux"|"mac"|"win" 
+
+FLAGS=
```

Comment:
Is this needed?

Reply:
The variable is used elsewhere in the script - shortly above this line.  While the value doesn't yet hold anything, I consider it a declaration of sorts, and it parallels /web/unit_tests's own test.sh.